### ISSUE TITLE
Auto publish master

### ISSUE THREAD

**Issue by stefanbuck** [OctoLinker/OctoLinker#437]
Our release process is a bit awkward. We do releases on a whenever-needed basis. This works, but it delays every new feature we add from a few days to a couple of weeks. I asked myself "Why not auto publishing changes in master straight away?". For publishing a new version we already use [chrome-webstore-upload-cli](https://github.com/DrewML/chrome-webstore-upload-cli) and there is workaround for Firefox https://github.com/mozilla/web-ext/issues/804#issuecomment-302588357

Outstanding questions:

- Do we need a [skip publish] - similar to [skip ci] to skip publishing a commit but run the CI?
- How do we proceed with the release notes?
- Should we drop the "in app" notification?
- Promote Opera version via https://addons.opera.com/en/extensions/details/download-chrome-extension-9/ like [github-issue-link-status](https://github.com/bfred-it/github-issue-link-status/tree/baa1262886c60949ccf679795516a313c2ee568a#installation-) does.

**Comment by josephfrazier** [comment]
I like the general idea, but I think we can tweak it a little bit to make the release process even easier.

>  I asked myself "Why not auto publishing changes in master straight away?".

I'm a little worried that this could result in updates that are *too* frequent? What if we kept the same release-on-version-bump system, but recommended [Chrome Store Foxified](https://addons.mozilla.org/en-US/firefox/addon/chrome-store-foxified/) for Firefox users, much like the [Install Chrome Extensions](https://addons.opera.com/en/extensions/details/download-chrome-extension-9/) Opera extension you mentioned? This would allow us to work around the Firefox/Opera release processes, as well as avoiding the `[skip publish]` and release notes issues. One thing I'm not sure about is whether Firefox/Opera users would receive automatic updates, or if they would have to reinstall every new version manually.

**Comment by stefanbuck** [comment]
Thanks for your feedback. I played around with both wrapper and well the user experience isn't that great. Both require a bunch of permissions which will scare some people.  

<img width="352" alt="screen shot 2018-01-23 at 20 17 35" src="https://user-images.githubusercontent.com/1393946/35296336-ffe424b4-007b-11e8-957c-ab438decdf31.png">

On Firefox I wasn't be able to install the extension, because of the following error
<img width="348" alt="screen shot 2018-01-23 at 20 27 13" src="https://user-images.githubusercontent.com/1393946/35296296-e1b65b38-007b-11e8-94d4-b012bd28e8ca.png">

In Opera the option page doesn't work and I get this alert when installing the extension.

<img width="334" alt="screen shot 2018-01-23 at 20 23 41" src="https://user-images.githubusercontent.com/1393946/35296299-e5f35ca0-007b-11e8-9cbe-7bd8af4e61b3.png">

It's a shame that Firefox and Opera  doesn't have a publish API. Anyway, I looked up the browser stats in GA and found out that we have 209 (1.6%) opera users only. 

<img width="362" alt="screen shot 2018-01-23 at 21 51 49" src="https://user-images.githubusercontent.com/1393946/35300091-c269be94-0087-11e8-90ee-ad9db8b5326c.png">

Fun fact, the last two versions are still pending in their review queue. Luckily Firefox improved there process and they are pretty fast by now. However, after writing this comment I think we should continue with a "real" OctoLinker for Firefox version and think about not publish further versions for opera anymore.
